engine/classes/Elgg/Database/EntityTable.php

```diff
@@ -14,48 +15,120 @@
  * @since      1.10.0
  */
 class EntityTable {
+
 	/**
-	 * Global Elgg configuration
-	 * 
-	 * @var \stdClass
+	 * @var \Elgg\Config
+	 */
+	protected $config;
+
+	/**
+	 * @var \Elgg\Database
 	 */
-	private $CONFIG;
+	protected $db;
 
 	/**
 	 * Constructor
+	 *
+	 * @param \Elgg\Config   $config Config
+	 * @param \Elgg\Database $db     Database
 	 */
-	public function __construct() {
-		global $CONFIG;
-		$this->CONFIG = $CONFIG;
+	public function __construct(\Elgg\Config $config, \Elgg\Database $db) {
+		$this->config = $config;
+		$this->db = $db;
 	}
 
 	/**
 	 * Returns a database row from the entities table.
 	 *
+	 * @see entity_row_to_elggstar()
+	 * 
 	 * @tip Use get_entity() to return the fully loaded entity.
 	 *
 	 * @warning This will only return results if a) it exists, b) you have access to it.
 	 * see {@link _elgg_get_access_where_sql()}.
 	 *
 	 * @param int $guid The GUID of the object to extract
-	 *
 	 * @return \stdClass|false
-	 * @see entity_row_to_elggstar()
 	 * @access private
 	 */
-	function getRow($guid) {
-		
-	
+	public function getRow($guid) {
+
 		if (!$guid) {
 			return false;
 		}
-	
-		$guid = (int) $guid;
-		$access = _elgg_get_access_where_sql(array('table_alias' => ''));
-	
-		return _elgg_services()->db->getDataRow("SELECT * from {$this->CONFIG->dbprefix}entities where guid=$guid and $access");
+
+		$access = _elgg_get_access_where_sql([
+			'table_alias' => '',
+		]);
+
+		$sql = "SELECT * FROM {$this->db->getTablePrefix()}entities
+			WHERE guid = :guid AND $access";
+
+		$params = [
+			':guid' => (int) $guid,
+		];
+
+		return $this->db->getDataRow($sql, null, $params);
 	}
-	
+
+	/**
+	 * Adds a new row to the entity table
+	 *
+	 * @param \stdClass $row Entity base information
+	 * @return int|false
+	 */
+	public function insertRow(\stdClass $row) {
```

Comment:
Ok. I will keep in mind during future refactor.  For now, I just want to merge this and get back to some down to earth work.